src/reducers/breakpoints.js

```diff
@@ -61,6 +62,22 @@ function update(
     case "REMOVE_BREAKPOINT": {
       return removeBreakpoint(state, action);
     }
+
+    case "ADD_PRETTY_SOURCE": {
+      const { source, generatedSource, mappings } = action;
+
+      function updateLocation({ location }) {
```

Comment:
maybe we can move this out of the reducer, wanted a bit of feedback on it though if it would be too slow